### ISSUE TITLE
object det with DAT

### ISSUE THREAD

**Issue by zhaoxin111** [alibaba/easyrobust#8]
In the papaer, 
DAT regards the two sub-tasks as a whole, and attacks the overall detection loss to produce adversarial images.

Does it mean that the total loss is directly used to calculate the gradient？
In addition, in the adversarial training of the object detection task, are the relevant parameters of the adversarial training the same as those of the classification task？

**Comment by zhaoxin111** [comment]
![image](https://user-images.githubusercontent.com/9295202/209779148-43506951-1765-4a6b-ba22-da47b1c4b697.png)
In L2Step/LinfStep, the project function clamp the x_adv within 0-1. If the preprocessed input is between -1 and 1, should it been clamp to [-1,1]?

**Comment by vtddggg** [comment]
1. Yes, total loss is directly used to calculate the gradient.
2. Hyper-parameters of DAT can be shared among different tasks.
3. As the input interval has changed, $\epsilon$ and `step_size` also need to be changed.

**Comment by zhaoxin111** [comment]
> 1. Yes, total loss is directly used to calculate the gradient.
> 2. Hyper-parameters of DAT can be shared among different tasks.
> 3. As the input interval has changed, ϵ and `step_size` also need to be changed.

Regarding the third point, are there any rules for how to adjust these hyperparameters?

**Comment by vtddggg** [comment]
For example，assume $\epsilon=1/255$ when $x_{adv}$ lies in [0, 1]. For new $\hat{x}_{adv}$ lies in [-1, 1], which is normalized by `mean=[0.5,0.5,0.5], std=[0.5,0.5,0.5]`, the new $\hat{\epsilon}$ should be $\hat{\epsilon}=2\times \epsilon $. Same rules are for `step_size`.

**Comment by zhaoxin111** [comment]
> For example，assume ϵ=1/255 when xadv lies in [0, 1]. For new x^adv lies in [-1, 1], which is normalized by `mean=[0.5,0.5,0.5], std=[0.5,0.5,0.5]`, the new ϵ^ should be ϵ^=2×ϵ. Same rules are for `step_size`.

Thank you for your answer
